### ISSUE TITLE
ParameterAveragingOptimizer: support scheduler

### ISSUE THREAD

**Issue by justheuristic** [learning-at-home/hivemind#239]
- [ ] switch self.averager from DecentralizedAverager to TrainingAverager
  ( see parameters https://github.com/learning-at-home/hivemind/blob/27ea94e3f932fa2d355d8a55050298c6e732f35b/hivemind/client/averaging/training.py#L24-L26 )
- [ ] add .epoch paramter to track local progress
- [ ] add optional scheduler parameter
- [ ] make a background thread triggered by event on each .step (see [progress_reporter](https://github.com/learning-at-home/hivemind/blob/master/hivemind/optim/collaborative.py#L112-L113) for example)
   - [ ] AFTER each step, add +1 to local steps and trigger progress event for progress reporter
   - [ ] this thread should report (current steps, epoch) to collaboration
      - use f"{self.prefix}.progress" as key and self.averager.endpoint as subkey
   - [ ] fetch steps and epoch from all the peers (see [fetch_collaboration_state](https://github.com/learning-at-home/hivemind/blob/master/hivemind/optim/collaborative.py#L286))
   - [ ] if some peer has higher epoch than us, set to his epoch
   - [ ] if peers on latest epoch collectively accumulated T steps, self.epoch += 1
- [ ] update scheduler to current epoch
- [ ] save/load epoch with state_dict/local_state_dict
- [ ] if remote epoch was above current epoch by more than M (parameter, e.g. 3), [load_state_from_peers](https://github.com/learning-at-home/hivemind/blob/27ea94e3f932fa2d355d8a55050298c6e732f35b/hivemind/client/averaging/training.py#L111-L132)
- optionally change name from `simple.py` to something more appropriate, e.g. `asgd.py` `averaged_opt.py`, **your ideas

**Comment by justheuristic** [comment]
The functionality was originally implemented by @xtinkt  in #252 , then reintegrated by me in #400 

The exact behavior of ParamterAveragingOptimizer can be achieved as such:
```
hivemind.Optimizer(..., use_local_updates=True, target_batch_size=AVERAGE_EVERY_THIS_MANY_SAMPLES)
```

For more detailed explanation, please refer to
* https://learning-at-home.readthedocs.io/en/latest/modules/optim.html
* quickstart on optimizer - https://learning-at-home.readthedocs.io/en/latest/user/quickstart.html
